clients/drcachesim/tracer/func_trace.cpp

```diff
@@ -125,24 +127,56 @@ func_post_hook(void *wrapcxt, void *user_data)
     append_entry(drcontext, TRACE_MARKER_TYPE_FUNC_RETVAL, retval);
 }
 
+static app_pc
+get_pc_by_symbol(const module_data_t *mod, const char *symbol)
+{
+    // Try to find the symbol in the dynamic symbol table.
+    app_pc pc = (app_pc)dr_get_proc_address(mod->handle, symbol);
+    if (pc != NULL) {
+        NOTIFY(1, "dr_get_proc_address found symbol %s at pc=%p\n", EMPTY_IF_NULL(symbol),
```

Comment:
I think you took the check-for-null too far: it doesn't seem like `symbol` is allowed to be NULL, right?  It seems like if you're going to check for that it should be at the top of the function and you should fail, or it should be required with an assert at the top, or sthg.

Reply:
Done.